File: backend/SKyber/src/main/java/edu/cit/SKyber/Entity/JobListing.java
package edu.cit.SKyber.Entity;



import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class JobListing {
    @Id
    private String id;
    private String jobTitle;
    private String companyName;
    private String address; 
    private String description;
    private String applicationlink;  
    private String jobImage;
    private String employementType; 
}

File: frontend_mobile/app/src/main/java/com/example/skyber/dataclass/JobListing.kt
package com.example.skyber.dataclass

import android.os.Parcelable
import com.google.firebase.database.IgnoreExtraProperties
import kotlinx.parcelize.Parcelize

@Parcelize
@IgnoreExtraProperties
data class JobListing (
    val id: String = "",
    val jobTitle: String = "",
    val companyName: String = "",
    val address: String = "",
    val description: String = "",
    val applicationLink: String = "",
    val employementType: String = "",//Part-Time or Full-Time
    val jobImage: String? = null,
): Parcelable